packages/ds-ext/src/DiagramEditorProvider.ts

```diff
@@ -35,9 +37,14 @@ export class DiagramEditorProvider implements vscode.CustomEditorProvider<Diagra
     );
   }
 
+  dispose(): void {
+    this.duckDBStorage.close();
+  }
+
   constructor(private readonly context: vscode.ExtensionContext) {
-    const storage = new DiagramObserverStorage();
-    this.inputObserverController = new InputObserverController(storage);
+    const dbPath = path.join(vscode.workspace.workspaceFolders![0].uri.fsPath, 'diagram.db');
```

Comment:
I suggest we move this to `./datastory/executions.db` to indicate it being temporary/git ignorable and describe the content a tiny bit clearer